src/common.rs

```diff
@@ -1102,84 +1088,54 @@ pub static EMPTY_STRING: WString = WString::new();
 /// todo!("Maybe remove the box? It is only needed for get_bg_context.")
 pub type CancelChecker = Box<dyn Fn() -> bool>;
 
-/// Converts the narrow character string \c in into its wide equivalent, and return it.
-///
-/// The string may contain embedded nulls.
-///
-/// This function encodes illegal character sequences in a reversible way using the private use
-/// area.
-pub fn str2wcstring(inp: &[u8]) -> WString {
-    if inp.is_empty() {
+/// Encodes the bytes in `input` into a [`WString`], encoding non-UTF-8 bytes into private-use-area
+/// code-points. Bytes which would be parsed into our reserved PUA range are encoded individually,
+/// to allow for correct round-tripping.
+pub fn bytes2wcstring(mut input: &[u8]) -> WString {
```

Comment:
Let's add a changelog entry under "Deprecations and removed features".

I'm good with removing support for non-UTF-8 outright,
though we don't have any telemetry to be able to tell if this puts needless pressure on real users.
If we want to be extra safe, we can do a feature flag, which users can turn back off for one release.
But that flag will be removed anyway in a future release (unless we hear about use cases).
Either way is fine by me.